homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -434,6 +441,29 @@ def _get_beolink_jid(self, entity_id: str) -> str:
         # Return JID
         return cast(str, config_entry.data[CONF_BEOLINK_JID])
 
+    async def _async_update_sound_modes(
+        self, active_sound_mode: ListeningModeProps | ListeningModeRef | None = None
+    ) -> None:
+        """Update the available sound modes."""
+        sound_modes = await self._client.get_listening_mode_set()
+
+        if active_sound_mode is None:
+            active_sound_mode = await self._client.get_active_listening_mode()
+
+        # Add the key to make the labels unique (As labels are not required to be unique on B&O devices)
+        for sound_mode in sound_modes:
+            label = f"{sound_mode.name} ({sound_mode.id})"
```

Comment:
How does an ID look like?

Reply:
For the Beosound Level, the built-in sound modes have ID 1, 2, 3 and 4 and user-made IDs start at 100. This would make the `Optimal` sound mode's label `Optimal (1)`. 
So no huge numbers.